cwru_opencv_common/src/opencv_geometry_3d.cpp

```diff
@@ -185,38 +185,34 @@ namespace cv_3d
         double dirLength = norm(dir);
         std::vector< Point > cornersV;
 
-        // If the length of the projected cylinder is longer than the radius of the end,
-        // draw a trapezoid.
-        if (dirLength > radEst0)
+        // Draw a trapezoid.
+        dir *= (1/norm(dir));
+        Point2d radDir(-dir.y, dir.x);
+        Point2d c0 = pt0+radDir*radEst0;
+        Point2d c1 = pt1+radDir*radEst1;
+        Point2d c2 = pt1-radDir*radEst1;
+        Point2d c3 = pt0-radDir*radEst0;
+
+        cornersV.resize(4);
+        cornersV[0] = corners[0] = Point(c0.x, c0.y);
+        cornersV[1] = corners[1] = Point(c1.x, c1.y);
+        cornersV[2] = corners[2] = Point(c2.x, c2.y);
+        cornersV[3] = corners[3] = Point(c3.x, c3.y);
+
+        // Draw the trapezoid.
+        fillConvexPoly(inputImage, corners, 4, color, CV_AA);
+
+        // Calculate the position and radius of the two ends.
+        if (tips.needed())
         {
-            dir *= (1/norm(dir));
-            Point2d radDir(-dir.y, dir.x);
-            Point2d c0 = pt0+radDir*radEst0;
-            Point2d c1 = pt1+radDir*radEst1;
-            Point2d c2 = pt1-radDir*radEst1;
-            Point2d c3 = pt0-radDir*radEst0;
-
-            cornersV.resize(4);
-            cornersV[0] = corners[0] = Point(c0.x, c0.y);
-            cornersV[1] = corners[1] = Point(c1.x, c1.y);
-            cornersV[2] = corners[2] = Point(c2.x, c2.y);
-            cornersV[3] = corners[3] = Point(c3.x, c3.y);
-
-            // Draw the trapezoid.
-            fillConvexPoly(inputImage, corners, 4, color, CV_AA);
-
-            // Calculate the position and radius of the two ends.
-            if (tips.needed())
-            {
-                tips.create(6, 1, CV_64FC1);
-                Mat tipMat(tips.getMat());
-                tipMat.at<double>(0) = pt0.x;
-                tipMat.at<double>(1) = pt0.y;
-                tipMat.at<double>(2) = radEst0;
-                tipMat.at<double>(3) = pt1.x;
-                tipMat.at<double>(4) = pt1.y;
-                tipMat.at<double>(5) = radEst1;
-            }
+            tips.create(6, 1, CV_64FC1);
```

Comment:
@tipakorng I put this condition in for cases when the cylinder would appear to be a circle. Is that a case you are eliminating for now and do not have any desire to put back in?